opts/options.go

```diff
@@ -149,3 +148,11 @@ func BucketSize(bucketSize int) Option {
 		return nil
 	}
 }
+
+// DisableAutoBootstrap disables auto bootstrap on the dht
```

Comment:
So, all this option _actually_ does is skip the _initial_ bootstrap. We'll still auto-bootstrap.

Given that, really, we _only_ care about this in tests (as far as I know), we could:

1. Have some kind of hidden option that prevents us from starting the bootstrap loop.
2. Manually call `startBootstrap`.

(We could also not start bootstrapping till the user calls `Bootstrap` but one of my goals here was to remove that requirement so the DHT "just works" out of the box)

Reply:
This option does two things actually:

1. Skip the initial bootstrap
2. Disables the "bootstrap if RT size is below minimum threshold" feature as that causes some tests to fail as I've explained in the comment [here](https://github.com/libp2p/go-libp2p-kad-dht/pull/403/files#r341797495)

I agree that we only need to do this for tests for now. But, what's the harm in having a visible option to disable this should someone wish to run the Dht without this side effect ?

Reply:
Ah, I see. Yeah, that makes sense (although it's still a bit confusing as it doesn't _completely_ disable automatic bootstrap). But some good documentation should be enough.

Reply:
Hm. Actually, I think this is still too confusing and too weird of an option to expose in a public API. I can't think of a single (real) use-case for it and I can think of plenty of ways in which it can be used incorrectly.

If we're going to have a public option to disable automatic bootstrapping, it should _completely_ disable automatic bootstrapping. We can probably just have this option modify the bootstrap config to set the bootstrap intervals to 0 (and then special-case 0 to disable the ticker). That way, we'll _only_ bootstrap on request.